### PR TITLE
SQL: CREATE VIEW and misc improvements [fixes #183]

### LINKED ISSUE #183: CREATE VIEW from multiple tables
### Reproduce
```sql
create or replace table t0 (x int) with system versioning;
create or replace table t1 (y int);
create or replace view vt0 as select * from t0;
create or replace view vt1 as select * from vt0;
create or replace view vt2 as select * from t1, vt0;
select * from vt0 for system_time all;
select * from vt1 for system_time all;
select * from vt2 for system_time all;
ERROR 4068 (HY000): System Versioning required: `FOR SYSTEM_TIME` query
```
### Expected
No error in case of `vt2` as it is analogous to `vt1`.

### PR DESCRIPTION
**Pull request by midenok**
Please, review.

### CODE REVIEW

File: sql/sql_lex.cc [hunk 1]
```diff
@@ -7044,3 +7044,29 @@ bool LEX::sp_add_cfetch(THD *thd, const LEX_STRING &name)
...
+  {
+    fld->name= (char *) memdup_root(thd->mem_root, buf, len);
+  }
+  else
+  {
+    push_warning_printf(thd,
+      Sql_condition::WARN_LEVEL_WARN,
+      WARN_VERS_ALIAS_TOO_LONG,
+      ER_THD(thd, WARN_VERS_ALIAS_TOO_LONG),
+      table->alias, field_name, field_name);
```
Comment (kevgs) [inline sql/sql_lex.cc]: Maybe use `buf` instead of a second `field_name` here?

File: sql/sql_view.cc [hunk 2]
```diff
@@ -453,38 +453,81 @@ bool mysql_create_view(THD *thd, TABLE_LIST *views,
...
+
+    if (!table->table)
+      continue;
+
+    /* Implicitly add versioning fields if needed */
+    if (table->table->versioned())
+    {
+      Item *item;
+      List_iterator_fast<Item> it(select_lex->item_list);
+      bool add_start= true;
```
Comment (kevgs) [inline sql/sql_view.cc]: You do not check order of system field here while it's necessary.
```sql
create view vt as select a, sys_trx_end, sys_trx_start from t;
```

File: sql/sql_lex.cc (hunk 1 above)
Comment (midenok) [reply sql/sql_lex.cc]: Did you read `WARN_VERS_ALIAS_TOO_LONG` message? How will it look?

**Review by kevgs — CHANGES_REQUESTED** [review]
Add vers fields position check at `create view`.

File: sql/sql_view.cc (hunk 2 above)
Comment (midenok) [reply sql/sql_view.cc]: Why it is neccessary?

Comment (kevgs) [reply sql/sql_view.cc]: Try this test case. `SELECTS` returns nothing here, while it should return 1 row.
```sql
create table t (a int) with system versioning;
insert into t values (1);
create view vt as select a, sys_trx_end, sys_trx_start from t;
select *, sys_trx_start, sys_trx_end from vt;
```

Our code relies on particular vers field position and order. As I understand, standard requires it too.

Comment (midenok) [reply sql/sql_view.cc]: Did you put order dependency somewhere? Do you know where is this bug in code?

Comment (kevgs) [reply sql/sql_view.cc]: [Here it is.](https://github.com/tempesta-tech/mariadb_10.2/commit/5a6844c927af6fd3798e2fe146dcffd062426d57#diff-baac2b2856d07d7828e3890b8eb9d217R16949)

Comment (midenok) [reply sql/sql_view.cc]: Fixed.

File: sql/sql_lex.cc (hunk 1 above)
Comment (kevgs) [reply sql/sql_lex.cc]: All works fine. I suggest to add this test case:
```sql
create table t (a int) with system versioning;
create view vt as select a from t as longlonglonglonglonglonglonglong;
show create view vt;
```

File: sql/sql_view.cc (hunk 2 above)
Comment (kevgs) [reply sql/sql_view.cc]: I suggest to add a test case:
```sql
create table t (a int) with system versioning;
insert into t values (1);
update t set a=2;
create view vt as select a, sys_trx_end, sys_trx_start from t for system_time all;
select * from vt;
```

**Review by kevgs — APPROVED** [review]

File: sql/sql_lex.cc (hunk 1 above)
Comment (midenok) [reply sql/sql_lex.cc]: There are such test cases. Look for 'baba'.

**Comment by midenok** [comment]
Please, review again. Diff [between old and new](https://github.com/tempesta-tech/mariadb_10.2/compare/am/183-multi_sysfield...tempesta-tech:am/trunk) versions:
```
git diff am/183-multi_sysfield am/trunk
```

File: sql/sql_view.cc [hunk 3]
```diff
@@ -453,38 +453,125 @@ bool mysql_create_view(THD *thd, TABLE_LIST *views,
...
   }
 
-  /* Implicitly add versioning fields if needed */
-  {
-    TABLE_LIST *tl = tables;
-    while (tl && tl->is_view())
-      tl = tl->view->select_lex.table_list.first;
-    if (tl && tl->table)
+  { /* System Versioning begin */
+    TABLE_LIST *impli_table= NULL, *expli_table= NULL;
```
Comment (kevgs) [inline sql/sql_view.cc:457]: IMO `implicit_table` and `explicit_table` looks better.

File: sql/sql_view.cc [hunk 4]
```diff
@@ -453,38 +453,125 @@ bool mysql_create_view(THD *thd, TABLE_LIST *views,
...
+      /* Implicitly add versioning fields if needed */
+      Item *item;
+      List_iterator_fast<Item> it(select_lex->item_list);
+
+      DBUG_ASSERT(table->alias);
+      while ((item= it++))
+      {
+        if (item->real_item()->type() != Item::FIELD_ITEM)
+          continue;
+        Item_field *fld= (Item_field*) (item->real_item());
```
Comment (kevgs) [inline sql/sql_view.cc:501]: Extra parentheses.

**Pull request merged**